### ISSUE TITLE
Automatic scrolling through history doesn't work properly.

### ISSUE THREAD

**Issue by donaastor** [alacritty/alacritty#6913]
Something goes wrong when I scroll through old commands:

1. Type a really long command, long enough to cover the whole screen and more, for ex. `printf "aaaaaaaaaaaaaaa...aa\n"` (like 10k `a`'s).
2. Run the command
3. Now press the `Up` arrow key to invoke the last command
4. Now press the `Home` key to, for example, change `printf` to `echo`

What happens is that rendering gets really messed up (I can't even explain) and sometimes I can't even solve it with a simple `clear`. The screen remains full of `a`'s.

**Comment by kchibisov** [comment]
Could you provide basic information which was mentioned in the issue template, like operating system and `alacritty -vv` log? Which shell were you using during it, have you trait plain `sh`? Are you using plain alacritty or are you using `tmux` or some other multiplexer?

If you type `reset` is it getting erased or if you select such phantom characters do they get cleared?

**Comment by donaastor** [comment]
@kchibisov Yes, here it is:
linux: 6.1.6.arch1-1
alacritty log: 
```
[0.000000730s] [INFO ] [alacritty] Welcome to Alacritty
[0.000025280s] [INFO ] [alacritty] Version 0.11.0 (8dbaa0bb)
[0.000032160s] [INFO ] [alacritty] Running on X11
[0.000148260s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/penis/.config/alacritty/alacritty.yml"
[0.001136678s] [DEBUG] [alacritty] Loading "monospace" font
[0.007726374s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Lcd", lcd_filter: 1 }
[0.010035199s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Lcd", lcd_filter: 1 }
[0.012336224s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Italic, load_flags: TARGET_LIGHT, render_mode: "Lcd", lcd_filter: 1 }
[0.014713929s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: TARGET_LIGHT, render_mode: "Lcd", lcd_filter: 1 }
[0.016551705s] [DEBUG] [alacritty] Estimated scaling factor: 2
[0.016563335s] [DEBUG] [alacritty] Estimated window size: None
[0.016567375s] [DEBUG] [alacritty] Estimated cell size: 18 x 34
[0.053307556s] [INFO ] [alacritty] Running on AMD Radeon RX 6650 XT (navi23, LLVM 14.0.6, DRM 3.49, 6.1.6-arch1-1)
[0.053322036s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.056814368s] [INFO ] [alacritty] Display scale factor: 2
[0.056827578s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.062887925s] [INFO ] [alacritty] Cell size: 18 x 34
[0.062901825s] [INFO ] [alacritty] Padding: 0 x 0
[0.062905625s] [INFO ] [alacritty] Width: 800, Height: 600
[0.081987254s] [INFO ] [alacritty] PTY dimensions: 17 x 44
[0.083771291s] [INFO ] [alacritty] Initialisation complete
[0.084353489s] [DEBUG] [alacritty_terminal] New num_cols is 106 and num_lines is 62
[0.084988868s] [INFO ] [alacritty] Padding: 0 x 0
[0.084994808s] [INFO ] [alacritty] Width: 1920, Height: 2111
[77.006176912s] [DEBUG] [alacritty_terminal] [Unhandled CSI] action='p', params=[0], intermediates=[33]
[77.006223922s] [DEBUG] [alacritty_terminal] [Unhandled CSI] action='l', params=[3;4], intermediates=[63]
[100.025997091s] [INFO ] [alacritty] Goodbye
```
alacritty.yml:
```
#font:
#  size: 19.0
```

I am using `bash` and reset indeed solves this. How do I test `sh`?

**Comment by kchibisov** [comment]
Are you talking about alacritty's selection or some other selection? I have a feeling is that just bash gets messed up and you assume that selection is from alacritty, while in alacritty it's all from bash.

You can test sh by `/bin/sh`, but it'll likely be symlinked to `bash`, so it'll have sort of the same issue.

You could try `dash` to test this, but it sounds like `bash` is being broken. Be aware that `Up`/`Home` and such are handled by bash as well.

**Comment by kchibisov** [comment]
just fwiw, it does happen in bash for me(not in zsh though), but `clear` should solve it(it does for me), unless bash decided to `re-render` something again. If you can reliably repro the part you've said about `clear`.

I'd suggest to do what you've done, but run alacritty with the `alacritty --ref-test` option and if you see anything left after clear just close alacritty window hotkey.

The `--ref-test` option will leave 3 json files(post the content of `size.json` and delete all of them) and `alacritty.recording`. If you open a fresh terminal and `cat alacritty.recording` it should hopefully provide a state where you have artifacts on screen you're mentioning. Attach this file as a `txt` and the content of `size.json`.

**Comment by donaastor** [comment]
Sorry, I discovered that clear works for me too. I have probably appended clear to some other command and didn't realize when it didn't work. Now it works. I suppose recording the output as you've told me doesn't make sense now. About what selection I was talking about: I don't really know. I just did a Ctrl+C in geany and then I paste using Ctrl+Shift+V in alacritty. To select in alacritty I double click on the word and press Ctrl+Shift+C.
